modules/etcd-defrag.adoc

```diff
@@ -0,0 +1,144 @@
+----
++
+Based on the `IS LEADER` column of this output, the [x-]`https://10.0.199.170:2379` endpoint is the leader. Matching this endpoint with the output of the previous step, the pod name of the leader is `etcd-ip-10-0-199-170.example.redhat.com`.
+
+. Defragment an etcd member.
+
+.. Connect to the running etcd container, passing in the name of a pod that is _not_ the leader:
++
+[source,terminal]
+----
+$ oc rsh -n openshift-etcd etcd-ip-10-0-159-225.example.redhat.com
+----
+
+.. Unset the `ETCDCTL_ENDPOINTS` environment variable:
++
+[source,terminal]
+----
+sh-4.4# unset ETCDCTL_ENDPOINTS
+----
+
+.. Defragment the etcd member:
++
+[source,terminal]
+----
+sh-4.4# etcdctl --command-timeout=30s --endpoints=https://localhost:2379 defrag
+----
++
+.Example output
+[source,terminal]
+----
+Finished defragmenting etcd member[https://localhost:2379]
+----
++
+If a timeout error occurs, increase the value for `--command-timeout` until the command succeeds.
+
+.. Verify that the database size was reduced:
++
+[source,terminal]
+----
+sh-4.4# etcdctl endpoint status -w table --cluster
+----
++
+.Example output
+[source,terminal]
+----
++---------------------------+------------------+---------+---------+-----------+------------+-----------+------------+--------------------+--------+
+|         ENDPOINT          |        ID        | VERSION | DB SIZE | IS LEADER | IS LEARNER | RAFT TERM | RAFT INDEX | RAFT APPLIED INDEX | ERRORS |
++---------------------------+------------------+---------+---------+-----------+------------+-----------+------------+--------------------+--------+
+|  https://10.0.191.37:2379 | 251cd44483d811c3 |   3.4.9 |  104 MB |     false |      false |         7 |      91624 |              91624 |        |
+| https://10.0.159.225:2379 | 264c7c58ecbdabee |   3.4.9 |   41 MB |     false |      false |         7 |      91624 |              91624 |        | <1>
+| https://10.0.199.170:2379 | 9ac311f93915cc79 |   3.4.9 |  104 MB |      true |      false |         7 |      91624 |              91624 |        |
++---------------------------+------------------+---------+---------+-----------+------------+-----------+------------+--------------------+--------+
+----
+This example shows that the database size for this etcd member is now 41 MB as opposed the starting size of 104 MB.
+
+.. Repeat these steps to connect to each of the other etcd members and defragment them. Be sure to defragment the leader last.
++
+Wait at least one minute between defragmentation actions to allow the etcd pod to recover. Until the etcd pod recovers, the etcd member will not respond.
+
+. If any `NOSPACE` alarms were triggered due to the space quota being exceeded, clear them.
+
+.. Check if there are any `NOSPACE` alarms:
++
+[source,terminal]
+----
+sh-4.4# etcdctl alarm list
+----
++
+.Example output
+[source,terminal]
+----
+memberID:<member_id> alarm:NOSPACE
```

Comment:
Does it make sense to elaborate on what `<member_id>` is here?

Reply:
It's really just sample output. I debated whether to just show it, but ended up doing this in case it was sensitive. Maybe I'll do something like 123456789 instead. Thanks!